solr/modules/s3-repository/src/java/org/apache/solr/s3/S3OutputStream.java

```diff
@@ -200,6 +202,10 @@ public MultipartUpload(String uploadId) {
     }
 
     void uploadPart(ByteArrayInputStream inputStream, long partSize) {
+      if (aborted) {
```

Comment:
this could just return like `complete()` below. Any reason to error here and not there? It'd make the logic a bit simpler in `close()` as well.

Reply:
It's true that this isn't consistent with the logic added in `close`. I kind of like throwing the error instead of just ignoring things because it otherwise hides the fact that the component is being misused (an upload after an abort). The reality is that none of the error paths should happen with the current code. I'm more inclined to throw the ISE exception in the `compoete()` if `aborted` than going the other way around.

>It'd make the logic a bit simpler in close() as well.

Maybe it's simpler if I check for `aborted` in `close()` too and handled separately? let me push a change and see

Reply:
yeah that makes sense to me.